### ISSUE TITLE
[3일차] Model-Agnostic Meta-Learning for Fast Adaptation of Deep Networks (MAML)

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#18]
Domain Generalization key concept 중 하나가 MAML 을 활용한 방법이다.
선행학습 차, MAML 논문을 읽어보게 되었다.

https://arxiv.org/abs/1703.03400

인용 수가 엄청나다.

목표는 적은 sample 만으로 finetuning 을 해서 새로운 task에서 좋은 성능을 내겠다는 것이다. (few shot train)

![image](https://user-images.githubusercontent.com/16400591/92843416-ad7c7600-f41f-11ea-9354-c3495b474e1e.png)

여러 task 에 대한 gradient 가 L1, L2, L3 처럼 다를 텐데 이걸 이용해서 meta-learning 학습 후 theta1, theta2, theta3 로의 finetuning을 쉽게 만들어 준다는 것이다.

알고리즘을 더 살펴보자.

![image](https://user-images.githubusercontent.com/16400591/92851854-ff75c980-f428-11ea-8069-a70738aaa27c.png)

T 는 모든 테스크가 짬뽕된 데이터이다.
alpha, beta 는 모두 learning rate hyperparmeter.
거기서 batch만큼 sample 해서 gradient descent 로 theta 를 업데이트 시킨 theta'_i 를 얻어낸다.
theta'_i 로 부터 다시 theta 를 업데이트 시킨다.

이게 끝이다. 다른 시각자료로도 이해해 보자.

![image](https://user-images.githubusercontent.com/16400591/92851653-cb020d80-f428-11ea-8646-c04fc1a5f028.png)
![image](https://user-images.githubusercontent.com/16400591/92852168-59768f00-f429-11ea-83d6-7fbd681fa83f.png)
![image](https://user-images.githubusercontent.com/16400591/92852264-7dd26b80-f429-11ea-881d-ec8a35d5e162.png)
가장 마지막 이미지가 theta' 을 사용하고 있어서 (위 알고리즘과 notation이 달라서) 조금 헷갈리긴 하지만, 
meta learning 중인 target theta 라 생각해주면 편할 것 같다.
